### ISSUE TITLE
cannot run composition function with crossplane/xfn:v1.11.1

### ISSUE THREAD

**Issue by duizabojul** [crossplane/crossplane#3807]
### What happened?

After upgrading to v1.11.1 (using latest helm chart) to avoid [flapping issues](https://github.com/crossplane/crossplane/issues/3747), I got errors when i am running my functions: 

```
cannot compose resources: cannot run Composition Function pipeline:
cannot run function "xnf-api": cannot run container: rpc error: code =
Unknown desc = exit status 1: xfn: error: spark.Command.Run(): OCI
runtime error: exit status 1
```

I thought it was a bug in my container script but even if I forward stdin to stdout, I got the above error.

Reverting to `crossplane/xfn:v1.11.0` fix this error.

### How can we reproduce it?

Use composition functions with xfn container running `crossplane/xfn:v1.11.1`. 

### What environment did it happen in?
Crossplane version: 
`crossplane/xfn:v1.11.1`

Kubernetes:
EKS 1.24

**Comment by duizabojul** [comment]
I digged a bit in this issue and i found the working image is running crun 1.5 and the failing one crun 1.8

**Comment by negz** [comment]
@duizabojul Thanks for the report! Are you able to share your Composition Function? I seem to be able to run functions in general using v1.11.1:

```yaml
$ cat ~/xp/fnio.yaml |docker run -i --security-opt=seccomp=unconfined crossplane/xfn:v1.11.1 run -c /tmp negz/xfn-quotable:latest -
apiVersion: apiextensions.crossplane.io/v1alpha1
config:
  apiVersion: database.example.org/v1alpha1
  kind: Config
  metadata:
    name: cloudsql
  spec:
    version: POSTGRES_9_6
kind: FunctionIO
observed:
  composite:
    connectionDetails:
    - name: uri
      value: postgresql://db.example.org:5432
    resource:
      apiVersion: database.example.org/v1alpha1
      kind: XPostgreSQLInstance
      metadata:
        name: my-db
      spec:
        compositionSelector:
          matchLabels:
            provider: gcp
        parameters:
          storageGB: 20
      status:
        conditions:
        - status: true
          type: Ready
```

**Comment by negz** [comment]
As a side-note, it would be ideal to pin the `crun` version we're using.

**Comment by duizabojul** [comment]
I just make a http call to a service inside my cluster. I tested again to use 1.11.1 and my XRs sync all failed, came back to 1.11.0 and all went green with my xfn doing its job. I tested with your command and with another image doing only local stuff and everything went fine with 1.11.1 so maybe it is a network related issue. I hope I'm not bothering you with a misconfiguration on my side but again the only difference between working and failing in my cluster is a matter of switching crossplane/xfn version 😭 

```
FROM alpine
RUN apk add --update --no-cache curl

ENTRYPOINT [ "curl" ]
CMD [ "-X", "POST", "-H", "Content-Type: application/yaml", "--data-binary", "@-", "http://....svc.cluster.local:3000/..." ]
```

My function configuration is the following:

```
- name: xfn-api      
  type: Container
  container:
    image: public.ecr.aws/.../xfn-runner:0.0.1
    imagePullPolicy: IfNotPresent
    resources:
      limits:
        memory: 10Mi
        cpu: 30m
    timeout: 20s
    network:
      policy: Runner
  config:
    apiVersion: crossplane.airnity.net/v1alpha1
    kind: XfnConfig
    ...
```

**Comment by garreeoke** [comment]
I get this issue as well, but it doesn't work regardless of which tag of xfn I use. @negz Trying to use your example above and get the same results. Is it possibly environmental?

**Comment by Kidswiss** [comment]
I get this, too.

Interestingly using `yq -o=json '.' redisFunctionIO.yaml |docker run -i --security-opt=seccomp=unconfined crossplane/xfn:v1.11.1 run -c /tmp kidswiss/myfunc:latest -` it works. But as soon as I want to run it within Crossplane it fails.

Here's the code I tested with: https://github.com/Kidswiss/compositionFunctionsTest

For the composition to work fully the cluster needs a cert-manager installed.

**Comment by AtzeDeVries** [comment]
having the same issue as @duizabojul. Reverting to 1.11.0 seems to fix it for me.

**Comment by bobh66** [comment]
I'm curious if this is a regression caused by using a later version of crun in the xfn container.   If you deploy the 1.11.1 Crossplane chart but override the xfn.image.tag to 1.11.0 do you still see the error?

If that fixes it we should probably pin crun to a known-good version as well is investigate why it's failing on later versions of crun.

**Comment by duizabojul** [comment]
@bobh66 yes this is my exact setting, 1.11.1 chart with xfn.image.tag to 1.11.0

**Comment by jbw976** [comment]
>  EDIT: okay figured it out, the output needs to be YAML, not JSON.

@Kidswiss does this mean you were able to successfully run your XFN with v1.11.1+? I'm trying to figure how many folks v1.11.1+ is working for.  I have some hope since it was working for @negz with https://github.com/crossplane/crossplane/issues/3807#issuecomment-1450905399, but @bobh66 theory about new version of `crun` could have validity 🤔

**Comment by Kidswiss** [comment]
@jbw976 yeah the python and go version in my link are both working. Although the go version has some bug and creates a reconcilation loop. But that's not a problem of crossplane itself :)

I did a lot more testing with it last week and I found that the

```
cannot compose resources: cannot run Composition Function pipeline:
cannot run function "xnf-api": cannot run container: rpc error: code =
Unknown desc = exit status 1: xfn: error: spark.Command.Run(): OCI
runtime error: exit status 1
```

Happens if the function container exits with a non 0 exit status. So I feel like the error is slightly misleading. As far as I was able to see the container actually run but failed.
